velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -26,19 +26,32 @@
 namespace facebook::velox::functions::sparksql {
 
 namespace detail {
-Expected<std::shared_ptr<DateTimeFormatter>> getDateTimeFormatter(
-    const std::string_view& format,
-    DateTimeFormatterType type) {
-  switch (type) {
-    case DateTimeFormatterType::STRICT_SIMPLE:
-      return buildSimpleDateTimeFormatter(format, /*lenient=*/false);
-    case DateTimeFormatterType::LENIENT_SIMPLE:
-      return buildSimpleDateTimeFormatter(format, /*lenient=*/true);
-    default:
+struct DateTimeFormatterProvider {
+  FOLLY_ALWAYS_INLINE void init(const core::QueryConfig& config) {
```

Comment:
Replacing the 'init' method with a constructor could be better as a constructor ensures that the object is properly initialized when it is created, preventing the possibility of using an uninitialized object. Also, it ensures members are immutable after the construction.

Reply:
DateTimeFormatterProvider has to be init in `*Function::initialize()` with QueryConfig. Using constructor might be a bit complex and inefficient.

Reply:
> Using constructor might be a bit complex and inefficient.

Would you please remind me of the reason for the complexity? I assume we can create an object for DateTimeFormatterProvider in the initialize method like below. 

```
formatterProvider_  = std::make_shared<detail::DateTimeFormatterProvider>(config).

private:
 std::shared<detail::DateTimeFormatterProvider> formatterProvider_ = nullptr;
```